test/Main.purs

```diff
@@ -437,6 +437,19 @@ main = do
       Just false -> HM.lookup k mc === HM.lookup k my
       Nothing -> false === HM.member k my
 
+  log "any"
+  quickCheck' 10000 $ \(a :: Array (Tuple Int Int)) ->
+    let hm = HM.fromFoldable a
+        xs = HM.toArrayBy Tuple hm
+    in  (not (HM.any (\_ -> true) HM.empty)) &&
+        case A.head xs of
+          Nothing -> true
+          Just h  -> case A.last xs of
+            Nothing -> true
+            Just l  -> HM.any (\x -> x == snd h) hm
+                    && HM.any (\x -> x == snd l) hm
+                    && (not (HM.any (\_ -> false) hm))
+
```

Comment:
```suggestion

  log "any agrees with Foldable any"
  quickCheck' 10000 $ \(a :: Array (Tuple CollidingInt Int)) (f :: Int -> Boolean) ->
    let m = HM.fromArray a
        f' (Tuple _ v) = f v
    -- use array instance, so we still test something useful if any ever becomes a method on Foldable
    in  any f' a === HM.any f m

```
This finds a couple of mutations I tried, even in collision nodes. I guess that means I was too pessimistic about random testing in the other comment.